### ISSUE TITLE
Expose a py_proto_library target for build_event_stream.proto so that way the python programs outside the Bazel repo can consume the BEP

### ISSUE THREAD

**Issue by aprotyas** [bazelbuild/bazel#15874]
### Description of the feature request:

Make the `src/main/java/com/google/devtools/build/lib/buildeventstream/proto/build_event_stream.proto` file publicly visible. Alternatively, export the file with `exports_files()`.

### What underlying problem are you trying to solve with this feature?

I have a Python tool that needs to ingest the build event stream, so I'm trying to generate a Python protobuf library for the BES.

I'm using the `@com_github_grpc_grpc//py_proto_library` rule as such:

```bzl
py_proto_library(
    name = "build_event_stream",
    deps = ["@io_bazel//src/main/java/com/google/devtools/build/lib/buildeventstream/proto:build_event_stream_proto"],
)
```

However, since the proto file of interest is in an external workspace, the generated `build_event_stream_proto_pb2.py` file has the following content which can't be imported:

```py
from ...io_bazel.src.main.java.com.google.devtools.build.lib.buildeventstream.proto.build_event_stream_pb2 import *
```

---

The workaround to this is to declare a `filegroup` with the proto file of interest, manually build a local proto library, and then list that target as a dependency (I've verified that this approach works with public proto files such as `@io_bazel//src/main/protobuf:extra_actions_base.proto`).

```bzl
# This example should work, but doesn't since build_event_stream.proto is not publicly visible

py_proto_library(
    name = "build_event_stream",
    deps = [":build_event_stream_proto"],
)

proto_library(
    name = "build_event_stream_proto",
    srcs = [":build_event_stream_proto_file"],
)

filegroup(
    name = "build_event_stream_proto_file",
    srcs = ["@io_bazel//src/main/java/com/google/devtools/build/lib/buildeventstream/proto:build_event_stream.proto"],
)
```

Unfortunately, this workaround does not work for this case because `build_event_stream.proto` is not publicly visible.

### Which operating system are you running Bazel on?

Ubuntu 20.04.4 LTS

### What is the output of `bazel info release`?

release 4.2.2

### Have you found anything relevant by searching the web?

Nothing exactly relevant, but there are a bunch of questions about which `py_proto_library` rule to actually use (I've found 3 implementations...).

**Comment by comius** [comment]
cc @michaeledgar, BEP is under team-core, right?

**Comment by haxorz** [comment]
> cc @michaeledgar, BEP is under team-core, right?

Yes, it is, so yes we'd be the team responsible for making the proposed change.

But can someone on `team-OSS` (or someone responsible for how Bazel is used) comment on whether this sort of change is desirable in general? I reckon there are other source files in the Bazel repo that other repos would want to consume; what's the best practice for allowing them / not allowing them to do that?

**Comment by haxorz** [comment]
On second thought, there's already [`//src/main/java/com/google/devtools/build/lib/buildeventstream/proto:ild_event_stream_java_proto`](https://github.com/bazelbuild/bazel/blob/dcd85855a75789a4d7dc47254bf9163248598303/src/main/java/com/google/devtools/build/lib/buildeventstream/proto/BUILD#L18) with intentionally-public visibility. So that precedent makes me think we don't need to think about setting a precedent ourselves here.

Therefore I think the proposed change is very reasonable.

**Comment by comius** [comment]
> Nothing exactly relevant, but there are a bunch of questions about which py_proto_library rule to actually use (I've found 3 implementations...).

Bazel is using `py_proto_library` from `@com_google_protobuf`.
https://github.com/bazelbuild/bazel/blob/570c1f27ae8a6205c03971fcfc5ded8cc99f4bf1/src/main/protobuf/BUILD#L47

The implementation is not perfect, but using that one, should prevent any compatibility issues and it hopefully doesn't have problems crossing repo boundaries.

The bazel team plans to provide `py_proto_library` as a native rule in Bazel by the end of September 2022, which should become the default one.

**Comment by comius** [comment]
> [Edit]: I think I need to create py_proto_lib instances of these dependencies and list those in the dep attribute.

Yes, it seems so. I didn't know that `protoc/py_proto_library` needs targets transitively (because it's not using an aspect). In this respect `grpc/py_proto_library` is better, but looking at the implementation is not handling repository boundaries correctly. :(

**Comment by aprotyas** [comment]
Hi team, any updates on this? I'm a little blocked by this right now. Neither of the two popular `py_proto_library` rule implementations (`@com_github_grpc_grpc`, `@com_google_protobuf`), seem to expose the required target correctly.

**Comment by tpudlik** [comment]
I too became interested in how `py_proto_library` is coming along, and found it's now a PR: https://github.com/bazelbuild/rules_python/pull/832. Looks pretty close to going in!
